### ISSUE TITLE
'nuget update -self' only updates to version 4.6.2 on WSL - Ubuntu 18.04.1 and fails permanently afterwards

### ISSUE THREAD

**Issue by mi-hol** [NuGet/Home#7170]
sudo nuget update -self
Checking for updates from https://www.nuget.org/api/v2/.
Currently running NuGet.exe 2.8.7.
Updating NuGet.exe to 4.6.2.
Update successful.

but then every following nuget command fails;
```
mihol@W10D-RS4:~$ sudo nuget install EPPlus.Core
Could not load file or assembly or one of its dependencies.
  Could not load file or assembly or one of its dependencies.
  Could not load file or assembly or one of its dependencies.
  Could not load file or assembly or one of its dependencies.
  Could not load file or assembly or one of its dependencies.
System.AggregateException: One or more errors occurred. ---> System.IO.FileNotFoundException: Could not load file or assembly or one of its dependencies.
   --- End of inner exception stack trace ---
  at NuGet.CommandLine.Program.Initialize (NuGet.IFileSystem fileSystem, NuGet.CommandLine.IConsole console) [0x000ce] in <36d8eef4280843bbad2e04b497b5fab0>:0
  at NuGet.CommandLine.Program.MainCore (System.String workingDirectory, System.String[] args) [0x000e3] in <36d8eef4280843bbad2e04b497b5fab0>:0
---> (Inner Exception #0) System.IO.FileNotFoundException: Could not load file or assembly or one of its dependencies.
File name: 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'<---

---> (Inner Exception #1) System.IO.FileNotFoundException: Could not load file or assembly or one of its dependencies.
File name: 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'<---
```

OS version:
Windows; Microsoft Windows 10.0.17134
WSL: Linux W10D-RS4 4.4.0-17134-Microsoft #112-Microsoft Thu Jun 07 22:57:00 PST 2018 x86_64 x86_64 x86_64 GNU/Linux
mono --version
Mono JIT compiler version 4.6.2 (Debian 4.6.2.7+dfsg-1ubuntu1)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        LLVM:          supported, not enabled.
        GC:            sgen

## Detailed repro steps so we can see the same problem
https://github.com/JanKallman/EPPlus/issues/224#issuecomment-410526228

1. start WSL bash

2. install nuget as described in https://docs.microsoft.com/en-us/nuget/install-nuget-client-tools MacOS/Linux
3.  sudo nuget update -self
4. sudo nuget install EPPlus.Core
or any other nuget install or update command

### Verbose Logs
```
sudo nuget update -self  -verbosity detailed
[sudo] password for mihol:
Could not load file or assembly or one of its dependencies.
  Could not load file or assembly or one of its dependencies.
  Could not load file or assembly or one of its dependencies.
  Could not load file or assembly or one of its dependencies.
  Could not load file or assembly or one of its dependencies.
System.AggregateException: One or more errors occurred. ---> System.IO.FileNotFoundException: Could not load file or assembly or one of its dependencies.
   --- End of inner exception stack trace ---
  at NuGet.CommandLine.Program.Initialize (NuGet.IFileSystem fileSystem, NuGet.CommandLine.IConsole console) [0x000ce] in <36d8eef4280843bbad2e04b497b5fab0>:0
  at NuGet.CommandLine.Program.MainCore (System.String workingDirectory, System.String[] args) [0x000e3] in <36d8eef4280843bbad2e04b497b5fab0>:0
---> (Inner Exception #0) System.IO.FileNotFoundException: Could not load file or assembly or one of its dependencies.
File name: 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'<---

---> (Inner Exception #1) System.IO.FileNotFoundException: Could not load file or assembly or one of its dependencies.
File name: 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'<---

---> (Inner Exception #2) System.IO.FileNotFoundException: Could not load file or assembly or one of its dependencies.
File name: 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'<---

---> (Inner Exception #3) System.IO.FileNotFoundException: Could not load file or assembly or one of its dependencies.
File name: 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'<---

---> (Inner Exception #4) System.IO.FileNotFoundException: Could not load file or assembly or one of its dependencies.
File name: 'System.Net.Http, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'<---
```

**Comment by mi-hol** [comment]
could be related to https://github.com/dotnet/corefx/issues/31449

**Comment by mi-hol** [comment]
Seems caused by a bug in WSL that is already fixed in W10 insider preview Microsoft/WSL#3286

with workaround in mono/mono#9891 mono installed.

After mono installed correcty the bug is gone :)

```
sudo nuget update -self  -verbosity detailed
[sudo] password for mihol:
NuGet Version: 4.7.0.5148
Checking for updates from https://www.nuget.org/api/v2/.
GET https://www.nuget.org/api/v2/GetUpdates()?packageIds='NuGet.CommandLine'&versions='4.7.0'&includePrerelease=false&includeAllVersions=false&targetFrameworks=''&versionConstraints=''
Currently running NuGet.exe 4.7.0.
NuGet.exe is up to date.
```
